py/private/py_image_layer.bzl

```diff
@@ -44,40 +44,40 @@ default_layer_groups = {
     "interpreter": "\\\\.runfiles/.*python.*-.*/",
 }
 
-def _split_mtree_into_layer_groups(name, root, groups, group_names, **kwargs):
+def _split_mtree_into_layer_groups(name, root, strip_prefix, groups, group_names, **kwargs):
     mtree_begin_blocks = "\n".join([
-        'print "#mtree" >> "$(RULEDIR)/%s.%s.manifest.spec";' % (name, gn)
+        'print "#mtree" >> "$(RULEDIR)/{name}.{gn}.manifest.spec";'.format(name = name, gn = gn)
         for gn in group_names
     ])
 
     # When an mtree entry matches a layer group, it will be moved into the mtree
     # for that group.
     ifs = "\n".join([
         """\
-if ($$1 ~ "%s") {
-    print $$0 >> "$(RULEDIR)/%s.%s.manifest.spec";
+if ($$1 ~ "{regex}") {{
+    print $$0 >> "$(RULEDIR)/{name}.{gn}.manifest.spec";
     next
-}""" % (regex, name, gn)
+}}""".format(regex = regex, name = name, gn = gn)
         for (gn, regex) in groups.items()
     ])
 
     cmd = """\
-awk < $< 'BEGIN {
-    %s
-}
-{
+awk < $< 'BEGIN {{
+    {mtree_begin_blocks}
+}}
+{{
     # Exclude .whl files from container images
-    if ($$1 ~ ".whl") {
+    if ($$1 ~ ".whl") {{
         next
-    }
+    }}
     # Move everything under the specified root
-    sub(/^/, ".%s")
+    sub(/^{strip_prefix}/, ".{root}")
```

Comment:
Note: this line is the primary change. Here we append `strip_prefix` to the beginning matcher, so that it is taken out of the path.

Reply:
strip_prefix needs more than this i believe,  see an example logic. 

https://github.com/bazel-contrib/bazel-lib/blob/c6707f1de817296befc9c238ddb0c27ed43a7c46/lib/private/modify_mtree.awk#L3-L31